### PR TITLE
Advent calendar

### PR DESCRIPTION
**Pull request by alexbridge**

### CODE REVIEW

File: frontend/subscriptions.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { appWillStart$, configuration, TAB_BAR_PATTERNS_BLACK_LIST } from '@shopgate/engage/core';
+
+export default (subscribe) => {
+  subscribe(appWillStart$, () => {
+    // Extend a blacklist where tab bar is hidden
+    configuration.update(TAB_BAR_PATTERNS_BLACK_LIST, blacklist => [
+      '/advent-calendar',
```
Comment (Carsten89) [inline frontend/subscriptions.js]: please create a constants for it and use it in both places

File: frontend/helpers.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  return (start.getTime() - diff) / 1000;
+};
+
+/**
+ * @param {number} day .
+ * @return {Object}
+ */
+export const getCalendarDay = day => (
+  calendar.find(d => (
+    parseInt(d.day, 10) === parseInt(day, 10)
```
Comment (Carsten89) [inline frontend/helpers.js:96]: i don't see a "day" prop in calendar config of the testshop

File: frontend/pages/AdventCalendar/index.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      <div className={styles.content}>
+        <CloseButton />
+
+        {fallingSnow && <Snowfall />}
+
+        <div className={styles.image}>
+          <img src={image} alt="" />
+        </div>
+
+        {isCalendarAllowed() && (
```
Comment (Carsten89) [inline frontend/pages/AdventCalendar/index.jsx]: isCalendarAllowed could be stored in a var
not sure how smart js engines are in this case here

File: frontend/components/ComingSoon/index.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  }),
+};
+
+/**
+ * @returns {JSX}
+ */
+const ComingSoon = () => (
+  <Fragment>
+    <p className={styles.p}>
+      {'Too early message. Come back later in'}
```
Comment (Carsten89) [inline frontend/components/ComingSoon/index.jsx]: should be configurable

**Review by Carsten89 — CHANGES_REQUESTED** [review]
i think you should be able to open also days from the past. Lets ask felix/martin

File: frontend/components/Snowfall/index.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import Snowflake from './components/Snowflake';
+import { canvas } from './styles';
+
+/**
+ * @returns {JSX}
+ */
+const Snowfall = () => (
```
Comment (Carsten89) [inline frontend/components/Snowfall/index.jsx:8]: snowflakes appear sometimes in the middle of the screen and fall from there. 
for me they should always appear at top and fall down from there

also the design looks to me like the snowfall should only be an overlay for the calendar itself and not the complete page

File: frontend/helpers.js (hunk 2 above)
Comment (alexbridge) [reply frontend/helpers.js:96]: Added to test shop. Its needed since calendar can be configured only for N days and not for all 24 days

**Pull request merged**